htlcswitch/packet.go

```diff
@@ -54,6 +54,13 @@ type htlcPacket struct {
 	// encrypted with any shared secret.
 	localFailure bool
 
+	// linkFailure is non-nil if we have failed the HTLC on our incoming or
+	// outgoing link. The values contained in the linkFailure are used to notify
+	// detailed link failure events. If the wire failure reason is enriched with
+	// additional information which we do not expose to the network, linkFailure
+	// will be a ForwardingError which contains additional information.
+	linkFailure lnwire.FailureMessage
```

Comment:
Does this now have overlap with `localFailure`? (also failures occurring at our node)

Reply:
It does, I thought they were different because we set `linkFailure` for incoming failures as well but local failure is only used for local sends so can combine the two.